docs/concepts/callbacks.md

```diff
@@ -1,2 +1,249 @@
+
+- `minimum_delta`: The minimum amount of improvement that a model can have over the previous best model to be considered worthwhile, zero by default, meaning that the training will not stop early unless the performance starts to decrease
+- `patience`: The number of consecutive rounds without improvement before the training is stopped, two by default.
+- `baseline`: an optional parameter that is used to compare the model's score against instead of the best previous model when checking for improvement. This baseline does not get changed over the course of a run.
+
+## WiSEFTCallback
+
+WiSE-FT, proposed by Mitchell et al. in [Robust fine-tuning of zero-shot models](https://arxiv.org/abs/2109.01903),
+has been proven to be an effective way for fine-tuning models with a strong zero-shot capability,
+such as CLIP:
+
+> Large pre-trained models such as CLIP or ALIGN offer consistent accuracy across a range of data distributions when performing zero-shot inference (i.e., without fine-tuning on a specific dataset). Although existing fine-tuning methods substantially improve accuracy on a given target distribution, they often reduce robustness to distribution shifts. We address this tension by introducing a simple and effective method for improving robustness while fine-tuning: ensembling the weights of the zero-shot and fine-tuned models (WiSE-FT).
+
+```python
+import finetuner
+from finetuner.callback import WiSEFTCallback
+
+run = finetuner.fit(
+    model='clip-base-en',
+    ...,
+    loss='CLIPLoss',
+    callbacks=[WiSEFTCallback(alpha=0.5)]
+)
+```
+
+```{warning}
+It is recommended to use WiSEFTCallback when fine-tuning CLIP.
+We can not ensure it works for other types of models, such as ResNet or BERT.
+
+Please refer to {ref}`Apply WiSE-FT <wise-ft>` in the CLIP fine-tuning example for more information.
+```
+
+## WandBLogger
+
+Finetuner allows you to utilize Weights & Biases for experiment tracking and visualization.
+The `WandBLogger` uses Weights & Biases [Anonymous Mode](https://docs.wandb.ai/ref/app/features/anon) to track a Finetuner Run.
+
+```{admonition} Use WandBLogger together with EvaluationCallback
+:class: hint
+The WandBLogger will track the training loss, plus the evaluation loss if `eval_data` is not None.
+
+If you use EvaluationCallback together with WandBLogger, search metrics will be tracked as well.
+Such as `mrr`, `precision`, `recall`, etc.
+```
+
+```python
+import finetuner
+from finetuner.callback import WandBLogger, EvaluationCallback
+
+run = finetuner.fit(
+    ...,
+    callbacks=[
+        EvaluationCallback(
+            query_data='finetuner/tll-test-query-da',
+            index_data='finetuner/tll-test-index-da'
+        ),
+        WandBLogger(),
+    ]
```

Comment:
The callbacks are in the right order in this example but should we also explain that the evaluation callback has to come before the wandblogger in order for the evaluation metrics to be tracked?